internal/formats/syftjson/to_syft_model.go

```diff
@@ -42,6 +46,59 @@ func toSyftLinuxRelease(d model.LinuxRelease) *linux.Release {
 	}
 }
 
+func toSyftRelationships(doc *model.Document, catalog *pkg.Catalog, relationships []model.Relationship) []artifact.Relationship {
+	idMap := make(map[string]interface{})
+
+	for _, p := range catalog.Sorted() {
+		idMap[string(p.ID())] = p
+		for _, l := range p.Locations {
+			idMap[string(l.Coordinates.ID())] = l.Coordinates
+		}
+	}
+
+	for _, f := range doc.Files {
+		idMap[f.ID] = f.Location
+	}
+
+	var out []artifact.Relationship
+	for _, r := range relationships {
+		syftRelationship := toSyftRelationship(idMap, r)
+		if syftRelationship != nil {
+			out = append(out, *syftRelationship)
+		}
+	}
+	return out
+}
+
+func toSyftRelationship(idMap map[string]interface{}, relationship model.Relationship) *artifact.Relationship {
+	from, ok := idMap[relationship.Parent].(artifact.Identifiable)
+	if !ok {
+		log.Warnf("relationship mapping from key %s is not a valid artifact.Identifiable type: %+v", relationship.Parent, idMap[relationship.Parent])
+		return nil
+	}
+	to, ok := idMap[relationship.Child].(artifact.Identifiable)
+	if !ok {
+		log.Warnf("relationship mapping to key %s is not a valid artifact.Identifiable type: %+v", relationship.Child, idMap[relationship.Child])
+		return nil
+	}
+	typ := artifact.RelationshipType(relationship.Type)
+
+	switch typ {
```

Comment:
if this is intended to encapsulate all syft-expressible relationships it would be less brittle to have a helper that checks against all possible relationships from a shared collection (something like `artifacts.AllRelationships()` that returns a slice, or similar). This way as we add relationships this section naturally grows.

Reply:
This has been removed during some refactoring